### PR TITLE
Allow people to download nlogo versions of Tortoise models

### PR DESCRIPTION
**Pull request by mrerrormessage**
This is a way of saving changes made in Tortoise

### CODE REVIEW

**Comment by mrerrormessage** [comment]
I'm uploading this for review, the functionality works properly in my tests, but the style still needs a bit of tweaking. I know that I broke "multi-tab resizing" (see photo below) with something I did, but I have no idea what I did to break it, nor do I know how to fix it, given that I didn't change any existing styles, but just added new ones. I will continue futzing around with it, but I figured that one of the css wizards might be able to tell me. Additionally, there's a decently-sized refactor to be reviewed in `CompilerService`. If feedback is available by tomorrow morning, I will try to get changes made (or at least started). Otherwise, someone else can finish the issue pretty easily while I'm gone or I can finish it when I get back.
screen shot 2015-06-24 at 8 32 06 pm

**Comment by mrerrormessage** [comment]
The above defect appears when browsing master on Firefox, so I'll open a separate issue for it and test this feature in Chrome.

**Comment by mrerrormessage** [comment]
New [bug has been opened](https://github.com/NetLogo/Galapagos/issues/251). This is ready for review.

File: app/assets/javascripts/TortoiseJS/agent/editor.coffee [hunk 1]
```diff
@@ -15,7 +15,10 @@ window.EditorWidget = Ractive.extend({
     """
     <div class="netlogo-code-container netlogo-widget">
       {{# !readOnly }}
-        <button class="netlogo-widget" on-click="recompile">compile</button>
+        <div class="netlogo-editor-button-container netlogo-widget" id="editor-buttons">
+          <button class="netlogo-tab editor-buttons netlogo-editor-button" id="editor-buttons" on-click="recompile">Compile</button>
+          <button class="netlogo-tab editor-buttons netlogo-editor-button" id="editor-buttons" on-click="exportnlogo">Export to NetLogo...</button>
+        </div>
```
Comment (TheBizzle) [inline app/assets/javascripts/TortoiseJS/agent/editor.coffee]: [Duplicate IDs are bad.](http://programmers.stackexchange.com/a/127180)  On top of that, it doesn't look to me like these IDs are ever actually used for anything, anyway...?

**Comment by TheBizzle** [comment]
I'm completed my latest round of slamming this branch with my hammer.  Major changes that I made were splitting `CompilerService` into multiple pieces (since I thought it was getting too big for its britches), and moving the "Export nlogo" button to a different location.

I'd like for @mrerrormessage to give it a look before we merge it.

**Comment by mrerrormessage** [comment]
Looks good to me. Think this is ready to be merged in. Thanks for the cleanups!

**Pull request merged**